sdk/core/src/error/http_error.rs

```diff
@@ -1,14 +1,17 @@
-use crate::{from_json, headers, Response, StatusCode};
+use crate::{
+    content_type, from_json,
+    headers::{self, Headers},
+    Response, StatusCode,
+};
 use bytes::Bytes;
 use serde::Deserialize;
-use std::collections::HashMap;
 
 /// An unsuccessful HTTP response
 #[derive(Debug)]
 pub struct HttpError {
     status: StatusCode,
     details: ErrorDetails,
-    headers: std::collections::HashMap<String, String>,
+    headers: Headers,
```

Comment:
Perhaps more flexible since `Headers` implements it:
```suggestion
    headers: impl Into<std::collections::HashMap<String, String>>,
```